libevmasm/Instruction.h

```diff
@@ -26,11 +26,185 @@
+	DUP9,				///< copies the 9th highest item in the stack to the top of the stack
+	DUP10,				///< copies the 10th highest item in the stack to the top of the stack
+	DUP11,				///< copies the 11th highest item in the stack to the top of the stack
+	DUP12,				///< copies the 12th highest item in the stack to the top of the stack
+	DUP13,				///< copies the 13th highest item in the stack to the top of the stack
+	DUP14,				///< copies the 14th highest item in the stack to the top of the stack
+	DUP15,				///< copies the 15th highest item in the stack to the top of the stack
+	DUP16,				///< copies the 16th highest item in the stack to the top of the stack
+
+	SWAP1,			///< swaps the highest and second highest value on the stack
+	SWAP2,				///< swaps the highest and third highest value on the stack
+	SWAP3,				///< swaps the highest and 4th highest value on the stack
+	SWAP4,				///< swaps the highest and 5th highest value on the stack
+	SWAP5,				///< swaps the highest and 6th highest value on the stack
+	SWAP6,				///< swaps the highest and 7th highest value on the stack
+	SWAP7,				///< swaps the highest and 8th highest value on the stack
+	SWAP8,				///< swaps the highest and 9th highest value on the stack
+	SWAP9,				///< swaps the highest and 10th highest value on the stack
+	SWAP10,				///< swaps the highest and 11th highest value on the stack
+	SWAP11,				///< swaps the highest and 12th highest value on the stack
+	SWAP12,				///< swaps the highest and 13th highest value on the stack
+	SWAP13,				///< swaps the highest and 14th highest value on the stack
+	SWAP14,				///< swaps the highest and 15th highest value on the stack
+	SWAP15,				///< swaps the highest and 16th highest value on the stack
+	SWAP16,				///< swaps the highest and 17th highest value on the stack
+
+	LOG0,				///< Makes a log entry; no topics.
+	LOG1,				///< Makes a log entry; 1 topic.
+	LOG2,				///< Makes a log entry; 2 topics.
+	LOG3,				///< Makes a log entry; 3 topics.
+	LOG4,				///< Makes a log entry; 4 topics.
+
+	CREATE,				///< create a new account with associated code
+	CALL,				///< message-call into an account
+	CALLCODE,			///< message-call with another account's code only
+	RETURN,				///< halt execution returning output data
+	DELEGATECALL,		///< like CALLCODE but keeps caller's value and sender
+	CREATE2,			///< create new account with associated code at address `sha3(0xff + sender + salt + init code) % 2**160`
+	STATICCALL,			///< like CALL but disallow state modifications
+
+	REVERT,				///< halt execution, revert state and return output data
+	INVALID,			///< invalid instruction for expressing runtime errors (e.g., division-by-zero)
+	SELFDESTRUCT,		///< halt execution and register account for later deletion
+						///
+	MAX_INTERNAL_INSTRUCTION ///< Marker for largest internal instruction value
+};
+
 /// Virtual machine bytecode instruction.
-enum class Instruction: uint8_t
+enum class InstructionOpCode: uint8_t
```

Comment:
Do we really still need this one? The instruction info array can just contain the plain bytecode and the traits like ``inline bool isPushInstruction`` can all be easily implemented on top of the other enum, can't they? Not sure I've seen all remaining uses of this yet, though.

Reply:
Oh, good thought. Yeah, it is very very rarely used, I think only in Disassembly.

Reply:
The inline functions are already implemented on top of the internal one, true